cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -235,25 +247,33 @@ protected String getServiceName() {
    */
   private void initializeOffsets() throws InterruptedException {
     // Setup initial offsets
-    Set<Integer> partitions = config.getPartitions();
-    while (offsets.size() != partitions.size() && !stopped) {
-      for (int partition : partitions) {
-        Checkpoint checkpoint = checkpoints.get(partition);
-        if (checkpoint != null) {
-          // TODO: (CDAP-7684) Deal with wrong offset issue
-          offsets.put(partition, checkpoint.getNextOffset());
-          continue;
-        }
+    final Set<Integer> partitions = new HashSet<>(config.getPartitions());
 
-        try {
-          // If no checkpoint, fetch from the beginning.
-          offsets.put(partition, getLastOffset(partition, kafka.api.OffsetRequest.EarliestTime()));
-        } catch (Exception e) {
-          OUTAGE_LOG.info("Failed to get Kafka earliest offset in {}:{} for pipeline {}. Will be retried",
-                          config.getTopic(), partition, name);
-          TimeUnit.SECONDS.sleep(1);
-          break;
-        }
+    while (!partitions.isEmpty() && !stopped) {
+      final int partition = partitions.iterator().next();
+
+      try {
+        Retries.callWithRetries(new Retries.Callable<Void, IOException>() {
```

Comment:
We should keep retrying until we found offsets for all partitions or the pipeline is stopped.